crates/runtime/src/lib.rs

```diff
@@ -322,6 +323,13 @@ impl_runtime_apis! {
         fn get_hasher() -> Hasher {
             Starknet::get_system_hash().into()
         }
+
+        fn extrinsic_filter(xts: Vec<<Block as BlockT>::Extrinsic>) -> Vec<Transaction>{
```

Comment:
2 questions:
- could this be done entirely client side ?
- we should include `DeclareTransaction` and `DeployAccountTransaction` extrinsics as well, not only `InvokeTransaction`, right ?

Reply:
After discussion with @tdelabro, it's fine to keep this logic in the runtime. @tdelabro to document the rationale

Reply:
@ZePedroResende please see this comment and add some doc/comment

Reply:
> * we should include `DeclareTransaction` and `DeployAccountTransaction` extrinsics as well, not only `InvokeTransaction`, right ?

Regarding this @abdelhamidbakhta @tdelabro, is this correct ?

Reply:
Yes we should add the two other types of transactions